<a href="https://colab.research.google.com/github/Sandeshh16/Amazon-Prime-TV-Shows-and-Movies-EDA/blob/main/Amazon_prime_EDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Project Name**    - Amazon Prime TV Shows and Movies



# **Project Summary -**
The goal of this project is to perform exploratory data analysis (EDA) on the Amazon Prime Video dataset, focusing on understanding the content catalog of TV shows and movies available on the platform. The analysis uncovers trends, insights, and patterns related to content type, genre distribution, ratings, release years, and regional availability.

# **GitHub Link -**

Provide your GitHub Link here.

# **Problem Statement**
This dataset was created to analyze all shows available on Amazon Prime Video, allowing us to extract valuable insights such as:

Content Diversity: What genres and categories dominate the platform?
Regional Availability: How does content distribution vary across different regions?
Trends Over Time: How has Amazon Prime’s content library evolved?
IMDb Ratings & Popularity: What are the highest-rated or most popular shows on the platform?
By analyzing this dataset, businesses, content creators, and data analysts can uncover key trends that influence subscription growth, user engagement, and content investment strategies in the streaming industry.


**Write Problem Statement Here.**

#### **Define Your Business Objective?**

Answer Here.

# **General Guidelines** : -  

1.   Well-structured, formatted, and commented code is required.
2.   Exception Handling, Production Grade Code & Deployment Ready Code will be a plus. Those students will be awarded some additional credits.
     
     The additional credits will have advantages over other students during Star Student selection.
       
             [ Note: - Deployment Ready Code is defined as, the whole .ipynb notebook should be executable in one go
                       without a single error logged. ]

3.   Each and every logic should have proper comments.
4. You may add as many number of charts you want. Make Sure for each and every chart the following format should be answered.
        

```
# Chart visualization code
```
            

*   Why did you pick the specific chart?
*   What is/are the insight(s) found from the chart?
* Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

5. You have to create at least 20 logical & meaningful charts having important insights.


[ Hints : - Do the Vizualization in  a structured way while following "UBM" Rule.

U - Univariate Analysis,

B - Bivariate Analysis (Numerical - Categorical, Numerical - Numerical, Categorical - Categorical)

M - Multivariate Analysis
 ]





#begin

In [ ]:
#import libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [ ]:
#fetch data
data1 = pd.read_csv('/content/titles.csv.zip')

In [ ]:
data2 = pd.read_csv('/content/credits.csv.zip')

In [ ]:
data1.columns

Index(['id', 'title', 'type', 'description', 'release_year',
       'age_certification', 'runtime', 'genres', 'production_countries',
       'seasons', 'imdb_id', 'imdb_score', 'imdb_votes', 'tmdb_popularity',
       'tmdb_score'],
      dtype='object')

In [ ]:
data2.columns

Index(['person_id', 'id', 'name', 'character', 'role'], dtype='object')

In [ ]:
#merge 2 data on common column id
data = pd.merge(data1,data2, on = 'id')

In [ ]:
#lets print the merged data
data

id              title   type  \
0         ts20945  The Three Stooges   SHOW   
1         ts20945  The Three Stooges   SHOW   
2         ts20945  The Three Stooges   SHOW   
3         tm19248        The General  MOVIE   
4         tm19248        The General  MOVIE   
...           ...                ...    ...   
124342  tm1054116           Baikunth  MOVIE   
124343  tm1054116           Baikunth  MOVIE   
124344  tm1054116           Baikunth  MOVIE   
124345  tm1054116           Baikunth  MOVIE   
124346  tm1054116           Baikunth  MOVIE   

                                              description  release_year  \
0       The Three Stooges were an American vaudeville ...          1934   
1       The Three Stooges were an American vaudeville ...          1934   
2       The Three Stooges were an American vaudeville ...          1934   
3       During America’s Civil War, Union spies steal ...          1926   
4       During America’s Civil War, Union spies steal ...          1926   
...                                                   ...           ...   
124342  This story is about prevalent caste problem, e...          2021   
124343  This story is about prevalent caste problem, e...          2021   
124344  This story is about prevalent caste problem, e...          2021   
124345  This story is about prevalent caste problem, e...          2021   
124346  This story is about prevalent caste problem, e...          2021   

       age_certification  runtime  \
0                  TV-PG       19   
1                  TV-PG       19   
2                  TV-PG       19   
3                    NaN       78   
4                    NaN       78   
...                  ...      ...   
124342               NaN       72   
124343               NaN       72   
124344               NaN       72   
124345               NaN       72   
124346               NaN       72   

                                                   genres  \
0       ['comedy', 'family', 'animation', 'action', 'f...   
1       ['comedy', 'family', 'animation', 'action', 'f...   
2       ['comedy', 'family', 'animation', 'action', 'f...   
3       ['action', 'drama', 'war', 'western', 'comedy'...   
4       ['action', 'drama', 'war', 'western', 'comedy'...   
...                                                   ...   
124342                                ['family', 'drama']   
124343                                ['family', 'drama']   
124344                                ['family', 'drama']   
124345                                ['family', 'drama']   
124346                                ['family', 'drama']   

       production_countries  seasons     imdb_id  imdb_score  imdb_votes  \
0                    ['US']     26.0   tt0850645         8.6      1092.0   
1                    ['US']     26.0   tt0850645         8.6      1092.0   
2                    ['US']     26.0   tt0850645         8.6      1092.0   
3                    ['US']      NaN   tt0017925         8.2     89766.0   
4                    ['US']      NaN   tt0017925         8.2     89766.0   
...                     ...      ...         ...         ...         ...   
124342                   []      NaN  tt14331982         8.4        49.0   
124343                   []      NaN  tt14331982         8.4        49.0   
124344                   []      NaN  tt14331982         8.4        49.0   
124345                   []      NaN  tt14331982         8.4        49.0   
124346                   []      NaN  tt14331982         8.4        49.0   

        tmdb_popularity  tmdb_score  person_id           name      character  \
0                15.424         7.6      59401     Joe Besser            Joe   
1                15.424         7.6      31460     Moe Howard            Moe   
2                15.424         7.6      31461     Larry Fine          Larry   
3                 8.647         8.0      21174  Buster Keaton    Johnny Gray   
4                 8.647         8.0      28713    Marion M

In [ ]:
# Dataset rows and columns count
data.shape

(124347, 19)

In [ ]:
#Fetching all the columns of dataset
data.columns

Index(['id', 'title', 'type', 'description', 'release_year',
       'age_certification', 'runtime', 'genres', 'production_countries',
       'seasons', 'imdb_id', 'imdb_score', 'imdb_votes', 'tmdb_popularity',
       'tmdb_score', 'person_id', 'name', 'character', 'role'],
      dtype='object')

In [ ]:
# Dataset Info
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 124347 entries, 0 to 124346
Data columns (total 19 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   id                    124347 non-null  object 
 1   title                 124347 non-null  object 
 2   type                  124347 non-null  object 
 3   description           124256 non-null  object 
 4   release_year          124347 non-null  int64  
 5   age_certification     56707 non-null   object 
 6   runtime               124347 non-null  int64  
 7   genres                124347 non-null  object 
 8   production_countries  124347 non-null  object 
 9   seasons               8153 non-null    float64
 10  imdb_id               119044 non-null  object 
 11  imdb_score            118296 non-null  float64
 12  imdb_votes            118272 non-null  float64
 13  tmdb_popularity       124332 non-null  float64
 14  tmdb_score            114082 non-null  float64
 15  

##**Dataset Description**
The dataset is related to movies and TV shows available on Amazon Prime Video. It's formed by merging two datasets: titles.csv and credits.csv.

**title.csv** dataset contains information about movies and TV shows available on Amazon Prime Video. Each row represents a single title.

**credits.csv** dataset contains information about the people involved in the production of the titles (actors, directors, etc.). Each row represents a person's involvement in a specific title.

###**Dataset has following columns**


**id**: Unique identifier for each title (movie or TV show)

**title**: Title of the movie or TV show

**type**: Whether it's a 'SHOW' (TV show) or a 'MOVIE'

**description**: A brief description of the title

**release_year**: The year the title was released

**age_certification**: Age rating or certification (e.g., PG-13, TV-MA)

**runtime**: Duration of the movie or TV show in minutes

**genres**: List of genres associated with the title (e.g., ['comedy', 'drama'])

**production_countries**: List of countries where the title was produced

**seasons**: Number of seasons (for TV shows only)

**imdb_id**: IMDb identifier for the title

**imdb_score**: IMDb rating of the title

**imdb_votes**: Number of votes the title received on IMDb

**tmdb_popularity**: Popularity score from TMDb (The Movie Database)

**tmdb_score**: Rating score from TMDb

**person_id**: Unique identifier for a person involved in the production (actor, director, etc.)

**name**: Name of the person (actor, director, etc.)

**character**: The character played by the person (for actors)

**role**: The role of the person in the production (e.g., 'ACTOR', 'DIRECTOR')

##**Missing values/Null values**

In [ ]:
#check for null values
data.isnull().sum()

id                           0
title                        0
type                         0
description                 91
release_year                 0
age_certification        67640
runtime                      0
genres                       0
production_countries         0
seasons                 116194
imdb_id                   5303
imdb_score                6051
imdb_votes                6075
tmdb_popularity             15
tmdb_score               10265
person_id                    0
name                         0
character                16307
role                         0
dtype: int64

we have null values in description column, Let's drop all the null values from description columns.

In [ ]:
data.dropna(subset = ['description'], inplace = True)

In [ ]:
data['age_certification'].mode()[0]

'R'

In [ ]:
#Replace all null values of age_certification with mode
data['age_certification'].fillna(data['age_certification'].mode()[0], inplace = True)

In [ ]:
data.isnull().sum()

id                           0
title                        0
type                         0
description                  0
release_year                 0
age_certification            0
runtime                      0
genres                       0
production_countries         0
seasons                 116155
imdb_id                   5234
imdb_score                5980
imdb_votes                6004
tmdb_popularity             13
tmdb_score               10239
person_id                    0
name                         0
character                16258
role                         0
dtype: int64

The 'seasons' column has null values because it's only applicable to TV shows, Movies don't have seasons so these entries would naturally be null. Instead of removing these rows, replace them with 0.

In [ ]:
data['seasons'].fillna(0, inplace = True)

In [ ]:
data.isnull().sum()

id                          0
title                       0
type                        0
description                 0
release_year                0
age_certification           0
runtime                     0
genres                      0
production_countries        0
seasons                     0
imdb_id                  5234
imdb_score               5980
imdb_votes               6004
tmdb_popularity            13
tmdb_score              10239
person_id                   0
name                        0
character               16258
role                        0
dtype: int64

Since 'imdb_id' is unique for Movies and TV shows, we will drop all null entries in 'imdb_id' columns.

In [ ]:
data.dropna(subset = ['imdb_id'], inplace = True)

In [ ]:
data.isnull().sum()

id                          0
title                       0
type                        0
description                 0
release_year                0
age_certification           0
runtime                     0
genres                      0
production_countries        0
seasons                     0
imdb_id                     0
imdb_score                746
imdb_votes                770
tmdb_popularity            13
tmdb_score               9118
person_id                   0
name                        0
character               14989
role                        0
dtype: int64

'imdb_score' represents the rating of a Movie or TV show, removing null values in 'imdb_score' might lead to data loss it's better to replace them with mean.

In [ ]:
round(data['imdb_score'].mean(),1)

np.float64(6.0)

In [ ]:
data['imdb_score'].fillna(round(data['imdb_score'].mean(),1), inplace = True)

In [ ]:
data.shape

(119022, 19)

'imdb_votes' represents the number of votes a Movie or TV show has recieved. Missing values in 'imdb_votes' may indicate movie is relatively new or has not gained much attention. It's better to replace the null values with 0.

In [ ]:
data['imdb_votes'].fillna(0, inplace = True)

In [ ]:
data.isnull().sum()

id                          0
title                       0
type                        0
description                 0
release_year                0
age_certification           0
runtime                     0
genres                      0
production_countries        0
seasons                     0
imdb_id                     0
imdb_score                  0
imdb_votes                  0
tmdb_popularity            13
tmdb_score               9118
person_id                   0
name                        0
character               14989
role                        0
dtype: int64

In [ ]:
# replace null values in 'tmdb_popularity' column with mean
round(data['tmdb_popularity'].mean(),2)

np.float64(10.32)

In [ ]:
data['tmdb_popularity'].fillna(round(data['tmdb_popularity'].mean(),2), inplace =True)

In [ ]:
# replace null values in 'tmdb_score' column with mean
round(data['tmdb_score'].mean(),1)

np.float64(6.0)

In [ ]:
data['tmdb_score'].fillna(round(data['tmdb_score'].mean(),1), inplace = True)

Missing values in the 'characters' column indicate that the information about the character played by a person in movie or TV show is not available in the dataset. So replace them by 'unknown'

In [ ]:
data['character'].fillna('unknown', inplace = True)

In [ ]:
data.isnull().sum()

id                      0
title                   0
type                    0
description             0
release_year            0
age_certification       0
runtime                 0
genres                  0
production_countries    0
seasons                 0
imdb_id                 0
imdb_score              0
imdb_votes              0
tmdb_popularity         0
tmdb_score              0
person_id               0
name                    0
character               0
role                    0
dtype: int64

In [ ]:
data.shape

(119022, 19)

##**Check for Duplicate values**

In [ ]:
data.duplicated().sum()

np.int64(168)

We have 168 duplicates in dataset, we will drop all the duplicate values

In [ ]:
data.drop_duplicates(inplace = True)

In [ ]:
data.duplicated().sum()

np.int64(0)

In [ ]:
data.dtypes

id                       object
title                    object
type                     object
description              object
release_year              int64
age_certification        object
runtime                   int64
genres                   object
production_countries     object
seasons                 float64
imdb_id                  object
imdb_score              float64
imdb_votes              float64
tmdb_popularity         float64
tmdb_score              float64
person_id                 int64
name                     object
character                object
role                     object
dtype: object

As we can see dataset is clean and ready for further analysis, The manipulations performed on given data are as follows,
1. **Data Loading and Merging** : Two datasets, titles.csv and credits.csv, were loaded and merged based on the common column 'id'.
2. **Handling Missing Values** :
- Null values in the 'description' and 'imdb_id' columns were dropped.
- Null values in 'age_certification' were replaced with the mode.
- Null values in 'seasons', 'imdb_votes', and 'character' were replaced with 0, 0, and 'unknown', respectively.
- Null values in 'imdb_score', 'tmdb_popularity', and 'tmdb_score' were replaced with their respective means.
3. **Handling Duplicates** : Duplicate rows identified and removed from the dataset.

#**Data Vizualization**

Data visualization is the graphical representation of information and data. By using visual elements like charts, graphs, and maps, data visualization tools provide an accessible way to see and understand trends, outliers, and patterns in data.

##Univariate Analysis

1. Box plot of imdb_score

2. histogram
- release_year
- runtime

3. Bar plot/count plot
- type(TV show or Movie)
- production_countries
- agr_certification

##Bivariate Analysis

1. scatter plot
- You might want to see if there's a relationship between imdb_score and tmdb_popularity (to check if higher-rated titles are also more popular).
- You might want to check the relationship between release_year and runtime of the title.
- You might want to see if there's a relationship between imdb_score and imdb_votes.

2. line plot
**release_year vs. imdb_score**
Reasoning: To visualize the trend of average IMDb scores of titles over the years. This can help you identify if movie/show quality (as reflected by IMDb scores) has been improving or declining over time.

**release_year vs. tmdb_popularity**
Reasoning: To track the trend of average TMDb popularity of titles over the years. This could reveal if movies/shows have become more or less popular over time.

**release_year vs. runtime**
Reasoning: To analyze how the average runtime of titles has changed over the years. This could show if movies/shows have been getting longer or shorter.

**seasons vs. imdb_score (for TV Shows only)**
Reasoning: To visualize the trend of IMDb scores for TV shows with different numbers of seasons. This can help you understand if shows with more seasons tend to be rated higher or lower.

**seasons vs. tmdb_popularity (for TV Shows only)**
Reasoning: To track the trend of TMDb popularity for TV shows with different numbers of seasons. This could reveal if shows with more seasons tend to be more or less popular.

## Multivariable Analysis


1. Pair plot

2. Heatmap